Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -256,17 +256,18 @@ bool parse_it(Iterator &first, Iterator last, RDKit::RWMol &mol) {
   if (first >= last || *first != '|') return false;
   ++first;
   while (first < last && *first != '|') {
+    Iterator::difference_type length = std::distance(first, last);
```

Comment:
This needs to be:
`typename Iterator::difference_type length = std::distance(first, last);`

I think that should also work with VS2008